python/tchannel/tornado/connection.py

```diff
@@ -75,13 +76,18 @@ class TornadoConnection(object):
     CALL_REQ_TYPES = frozenset([Types.CALL_REQ, Types.CALL_REQ_CONTINUE])
     CALL_RES_TYPES = frozenset([Types.CALL_RES, Types.CALL_RES_CONTINUE])
 
-    def __init__(self, connection):
+    def __init__(self, connection, tchannel=None):
         assert connection, "connection is required"
 
         self.closed = False
         self.connection = connection
+        try:
+            (self.remote_host,
+             self.remote_host_port) = connection.socket.getsockname()
```

Comment:
The remote_host is set correctly when we receive a handshake. We expect the information in the handshake. It includes both, the address and the port. So, for a tornado connection, remote_host is the address (with port) on which you can reach that service.